### DISCUSSION TITLE
Questions about Tracing Sub-segments in Private API Using Lambda PowerTools

### DISCUSSION THREAD

**Discussion by jaeyoung0509** [aws-powertools/powertools-lambda-python#3279] [General]
Firstly, I've been effectively using Lambda PowerTools for my serverless development. 
In my setup, I have a Lambda service that acts as an API Gateway `Public API`, and beyond that, there's a  `Private API` Lambda which carries out the actual logic. 
I aim to trace the request flow as: 
`request -> public API Lambda(*root segement)-> private API Lambda(*sub segment) -> aws service`

I've perused the documentation and noted that using powertools.trace automatically generates a trace header. I can forward this header to the private API Lambda. However, I'm uncertain about how to register a sub-segment for this private API Lambda.

**Comment by leandrodamascena** [comment]
Hello @jaeyoung0509! Firstly, sorry for the late response, we normally respond in less than 1 day but this was an unusual week where we missed some responses on time.

Let me see if I understand your user case:
In the internet-facing API with Lambda you have the root segment with `lambda_handler` and then you invoke a Private API with Lambda and want to create only a subsegment and not a root segment? I'm sorry, but I don't think I'm understanding the whole picture here.

Do you think you can draw a flow diagram or send some snippets/screenshots? This would make my understanding easier.

Thank you so much and I will put this discussion in my priorities.
